### PR TITLE
[WIP] v2.4.0

### PR DESCRIPTION
**Pull request by N3TC4T**

### CODE REVIEW

**Comment by vixentael** [comment]
i leave my comment here to follow commits

**Review by vixentael — COMMENTED** [review]
👀 👍

File: ios/XUMM/Libs/Security/Vault/VaultManager.m [hunk 1]
```diff
@@ -189,7 +189,7 @@ + (BOOL) reKeyVault: (NSString *)vaultName oldKey:(NSString *)oldKey newKey:(NSS
   // try to create the new vault under a temp name with new key
   // with this we will make sure we are able to recover the key in case of failure
   NSString *recoveryVaultName = getRecoveryVaultName(vaultName);
-  [VaultManagerModule createVault:recoveryVaultName data:cleartext key:newKey];
+  [VaultManagerModule createVault:recoveryVaultName data:cleartext key:oldKey];
```
Comment (vixentael) [inline ios/XUMM/Libs/Security/Vault/VaultManager.m:192]: 😬😅

Comment (N3TC4T) [reply ios/XUMM/Libs/Security/Vault/VaultManager.m:192]: Haha yeah, I forgot to change the comment

**Pull request merged**